app/exerlog/lib/UI/login_screen/login_page.dart

```diff
@@ -45,140 +41,127 @@ class _LoginPageState extends State<LoginPage> {
-                  onPressed: () async {
-                    if (index == 0) {
-                      // login with email and password
-                      if (loginData.password != '' && loginData.email != '') {
-                        User? user = await EmailSignup.signInWithEmailAndPassword(loginData.email, loginData.password).then((value) {
-                          userID = value?.uid;
-                          return value;
-                        });
-                       
-                
-                if (user != null) {
-                  print("USER IS NOT NULL");
-                  userID = user.uid;
-                  the_user = user;
-                  setState(() {
-                    Navigator.pop(context);
-                    Navigator.of(context).push(
-                    MaterialPageRoute(
-                      builder: (context) => WorkoutPage(null
-                      ),
+                ),
+                Padding(
+                  padding: const EdgeInsets.fromLTRB(60, 0, 60, 16),
+                  child: RaisedGradientButton(
+                    radius: 30,
+                    child: Text(
+                      index > 0 ? "Sign up" : "Login",
+                      style: buttonText,
                     ),
-                  );
-                  });                 
-                }
-                      }
-                    }
-                    else {
-                      // signup with email and password
-                      if (loginData.password != '' && loginData.email != '') {
-                        User? user = await EmailSignup.registerWithEmail(loginData.email, loginData.password).then((value) {
-                          userID = value?.uid;
-                          return value;
-                        });
-                       
-                
-                if (user != null) {
-                  print("USER IS NOT NULL");
-                  userID = user.uid;
-                  the_user = user;
-                  setState(() {
-                    Navigator.pop(context);
-                    Navigator.of(context).push(
-                    MaterialPageRoute(
-                      builder: (context) => WorkoutPage(null
-                      ),
+                    gradient: LinearGradient(
+                      colors: <Color>[Color(0xFF34D1C2), Color(0xFF31A6DC)],
                     ),
-                  );
-                  });                 
-                }
+                    onPressed: () async {
+                      if (index == 0) {
+                        // login with email and password
+                        if (loginData.password != '' && loginData.email != '') {
+                          final user = await EmailSignup.signInWithEmailAndPassword(
+                            loginData.email,
+                            loginData.password,
+                          );
+
+                          if (user != null) {

```

Comment:
I personally would do a guard clause, easier to read:

```dart
if (user == null) {
    show error message to user
    return
}

whatever needs to happen after signIn...
```

Reply:
I limited myself to make only UI changes there, all the logic is still the same as it was before
But I agree with you, there's a lot of room for improvement there